src/libraries/System.Diagnostics.Process/tests/ProcessStreamReadTests.cs

```diff
@@ -491,6 +522,47 @@ public void TestManyOutputLines()
             Assert.Equal(ExpectedLineCount + 1, totalLinesReceived);
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void TestManyOutputLines_SynchronizingObject(bool invokeRequired)
+        {
+            const int ExpectedLineCount = 144;
+
+            int nonWhitespaceLinesReceived = 0;
+            int totalLinesReceived = 0;
+            int invokeCalled = 0;
+
+            Process p = CreateProcessPortable(RemotelyInvokable.Write144Lines);
+            p.SynchronizingObject = new DelegateSynchronizeInvoke()
+            {
+                InvokeRequiredDelegate = () => invokeRequired,
+                InvokeDelegate = (d, args) =>
+                {
+                    invokeCalled++;
+                    return d.DynamicInvoke(args);
+                }
+            };
+            p.StartInfo.RedirectStandardOutput = true;
+            p.OutputDataReceived += (s, e) =>
+            {
+                if (!string.IsNullOrWhiteSpace(e.Data))
+                {
+                    nonWhitespaceLinesReceived++;
+                }
+                totalLinesReceived++;
+            };
+            p.Start();
+            p.BeginOutputReadLine();
+
+            Assert.True(p.WaitForExit(WaitInMS));
+            p.WaitForExit(); // This ensures async event handlers are finished processing.
+
+            Assert.Equal(ExpectedLineCount, nonWhitespaceLinesReceived);
+            Assert.Equal(ExpectedLineCount + 1, totalLinesReceived);
+            Assert.Equal(invokeRequired ? totalLinesReceived : 0, invokeCalled);
```

Comment:
nit: the `TestAsyncErrorStream_SynchronizingObject` tests both contents of the output and the number of invocations. This test tests only the number of invocations. It would be nice to test the content too

Reply:
Yeah, this is just a clone of the test above it, which doesn't validate it either.  I'm going to leave it as is for now.